internal/controller/buildpipeline/buildpipeline_adapter.go

```diff
@@ -602,16 +602,22 @@ func (a *Adapter) ReportIntegrationTestStatusAccordingToBuildPLR(pipelineRun *te
 	a.logger.Info("Reporter initialized", "reporter", reporter.GetReporterName())
 
 	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		var statusCode int
 		intgTestStatusDetails, err := generateIntgTestStatusDetails(pipelineRun, integrationTestStatus)
 		if err != nil {
 			return err
 		}
-		err = status.IterateIntegrationTestInStatusReport(a.context, a.client, reporter, snapshot, integrationTestScenarios, intgTestStatusDetails, componentName)
+		statusCode, err = status.IterateIntegrationTestInStatusReport(a.context, a.client, reporter, snapshot, integrationTestScenarios, intgTestStatusDetails, componentName)
 		if err != nil {
 			a.logger.Error(err, fmt.Sprintf("failed to report integration test status according to build pipelinerun %s/%s",
 				pipelineRun.Namespace, pipelineRun.Name))
-			return fmt.Errorf("failed to report integration test status according to build pipelineRun %s/%s: %w",
-				pipelineRun.Namespace, pipelineRun.Name, err)
+			if reporter.ReturnCodeIsUnrecoverable(statusCode) {
+				// We return nil in case the status code is not recoverable by retrying, e.g. when status is 403 unauthorised
```

Comment:
We may log out an error message before `return nil` then can have a chance to know it from splunk log.

Reply:
There is a [logger call just above this condition](https://github.com/konflux-ci/integration-service/pull/1199/files#diff-3540d1f95a378c2e6bea89135278f11b57b55c7c4e0a7ac892eb6dfd251b5b55R612) which logs out the error in all cases.

Reply:
Oh. Sorry, not notice it